R/conf_mat.R

```diff
@@ -274,3 +286,90 @@ summary.conf_mat <- function(object,
 
   stats
 }
+
+conf_mat_plot_types <- c("mosaic", "heatmap")
+
+# Dynamically exported
+#' @rdname conf_mat
+#' @param type Type of plot desired, must be "mosaic" or "heatmap",
+#'   defaults to "mosaic".
+#' @param object The `conf_mat` data frame returned from `conf_mat()`.
+autoplot.conf_mat <- function(object, type = "mosaic", ...) {
+
+  if (!(type %in% conf_mat_plot_types)) {
+    stop("`type` must be one of: ",
+         paste(conf_mat_plot_types, collapse = ", "), call. = FALSE)
+  }
+
+  switch(type,
+         mosaic = cm_mosaic(object),
+         heatmap = cm_heat(object))
+}
+
+
+
+cm_heat <- function(x) {
+
+  `%+%` <- ggplot2::`%+%`
+
+  data <- as.data.frame.matrix(x$table)
```

Comment:
Do you need `as.data.frame.matrix`? `x$table` is a `table` and `as.data.frame.table` (the default method here) seems to do what you try and do manually with the rownames and `gather()`

Reply:
you are right, `as.data.frame.table` does exactly what is needed